core/dive.c

```diff
@@ -573,6 +573,30 @@ void copy_events(const struct divecomputer *s, struct divecomputer *d)
 	*pev = NULL;
 }
 
+/* copies all events from all dive computers before a given time
+   this is used when editing a dive in the planner to preserve the events
+   of the old dive */
+void copy_events_until(const struct dive *sd, struct dive *dd, int time)
+{
+	if (!sd || !dd)
+		return;
+
+	const struct divecomputer *s = &sd->dc;
+	struct divecomputer *d = &dd->dc;
+
+	while (s && d) {
+		const struct event *ev;
+		ev = s->events;
+		while (ev != NULL) {
+			if (ev->time.seconds < time && !event_is_gaschange(ev))
+				add_event(d, ev->time.seconds, ev->type, ev->flags, ev->value, strdup(ev->name));
```

Comment:
Just one nitpick - this seems to leak memory: `add_event` copies the name (because the name is part of the event-structure). Thus, the `strdup` is redundant, isn't it.